src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -68,96 +51,153 @@ private Optional<EppoValue> getAssignmentValue(
         InputValidator.validateNotBlank(subjectKey, "Invalid argument: subjectKey cannot be blank");
         InputValidator.validateNotBlank(flagKey, "Invalid argument: flagKey cannot be blank");
 
+        VariationAssignmentResult assignmentResult = this.getAssignedVariation(flagKey, subjectKey, subjectAttributes);
+
+        if (assignmentResult == null) {
+            return Optional.empty();
+        }
+
+        Variation assignedVariation = assignmentResult.getVariation();
+        Optional<EppoValue> assignmentValue = Optional.of(assignedVariation.getTypedValue());
+
+        // Below is used for logging
+        String experimentKey = assignmentResult.getExperimentKey();
+        String allocationKey = assignmentResult.getAllocationKey();
+        String assignedVariationString = assignedVariation.getTypedValue().stringValue();
+
+        if (assignedVariation.getAlgorithmType() == AlgorithmType.OVERRIDE) {
+            // Assigned variation was from an override; return its value without logging
+            return assignmentValue;
+        } else if (assignedVariation.getAlgorithmType() == AlgorithmType.BANDIT) {
+            // Assigned variation is a bandit; need to use the bandit to determine its value
+            assignmentValue = this.determineAndLogBanditAction(assignmentResult, assignmentOptions);
+        }
+
+        // Log the assignment
+        try {
+            this.eppoClientConfig.getAssignmentLogger()
+                    .logAssignment(new AssignmentLogData(
+                            experimentKey,
+                            flagKey,
+                            allocationKey,
+                            assignedVariationString,
+                            subjectKey,
+                            subjectAttributes));
+        } catch (Exception e) {
+            log.warn("Error logging assignment", e);
+        }
+
+        return assignmentValue;
+    }
+
+    private VariationAssignmentResult getAssignedVariation(String flagKey, String subjectKey, EppoAttributes subjectAttributes) {
+
         // Fetch Experiment Configuration
         ExperimentConfiguration configuration = this.configurationStore.getExperimentConfiguration(flagKey);
         if (configuration == null) {
             log.warn("[Eppo SDK] No configuration found for key: " + flagKey);
-            return Optional.empty();
+            return null;
```

Comment:
Why the change to returning `null` ?

Reply:
This is within `getAssignedVariation()` now with the end goal to return a `VariationAssignmentResult`. I use `null` to basically indicate no assignment could be made (vs. always returning a result but with a null variation)